t.in.eoarchive.py

```diff
@@ -0,0 +1,612 @@
+                                scenedatetime_str = "-".join(scene.replace(
+                                    "-", "_").split("_")[1:3])
+                                scene_datetime = datetime.strptime(scenedatetime_str,
+                                                                   "%Y%m%d-%H%M%S")
+                                scene_dir["datetime"] = scene_datetime
+                                for bandsuffix, bandname in bands_filesuffixes.items():
+                                    # we need the whole <band>.tif string to rule out
+                                    # confusions between B8 and B8A
+                                    file_ending = f"{bandsuffix}{file_format}"
+                                    if bandname != "S2_CLM":
+                                        for file in bands_in_scenedir:
+                                            if file_ending in file:
+                                                scene_dir["band_paths"][bandname] = file
+                                    else:
+                                        for file in os.listdir(os.path.join(scene_path,
+                                                                            clm_dir)):
+                                            if file_ending in file:
+                                                scene_dir["band_paths"][bandname] = \
+                                                    os.path.join(clm_dir, file)
+                            scenes_to_import.append(scene_dir)
+    return scenes_to_import
+
+
+def import_raster(paramdict):
+    """ Imports a raster dataset, checks whether a map is empty
+        and adds a semantic label
+    Args:
+        paramdict(dict): Dictionary with import/labelling parameters.
+                         Required dictionary key value pairs are:
+                         - name(string): name of the imported raster map
+                         - input(string): path to the raster file to import
+                         - memory(int): allocated memory for the import process
+                         - semantic_label(string): semantic label to attach to
+                                                   imported raster
+                         - scene_datetime(datetime): timestamp of the raster map
+    Returns:
+        import_dict(dict): Dictionary with information of the imported map.
+                           Dictionary key value pairs are:
+                           - name(string): name of the imported raster map
+                           - datetime(datetime): timestamp of the raster map
+                           - map_empty(bool): indicated whether the map consists
+                                              of NoData only
+                           - band(string): band/semantic label attached to the
+                                           imported raster map
+    """
+    name = paramdict["name"]
+    input = paramdict["input"]
+    memory = paramdict["memory"]
```

Comment:
memory must be smaller than 100000, otherwise GDAL interprets it as bytes, not MB.

Reply:
So this means that if memory is larger than 100.000, we need to convert it to Bytes (i.e. memory * 1000000) to get the correct memory value (as implemented now)?

Reply:
Yes, or clip it to 100.000 MB, this should be far more than enough, even if split up later on during parallelization. Or adjust it later on for each parallel job, seems to be the better place.

Reply:
According to https://gdal.org/programs/gdalwarp.html , the threshold is already at 10000 MB, which on a heavy computing machine /and/or running only on few cores, may be a realistic value we want to pass as memory parameter to individual processes. 
It is now adjusted that the value is multiplied by 1000000 if there is to be more than 10000 (MB) of memory used for each process.

Is this threshold also somewhere noted in the GRASS manual? In every memory page it is only declared as memory (in MB)...

Reply:
This limit applies to the GDAL env var `GDAL_CACHEMAX`, see https://github.com/OSGeo/gdal/blob/master/gcore/gdalrasterblock.cpp#L279

Sorry for being unclear!

More details: if this memory option is passed via `r.import` to `r.in.gdal memory=<memory>`, `r.in.gdal` will use `CPLSetConfigOption("GDAL_CACHEMAX", parm.memory->answer);` where this limitation applies within GDAL.

Reply:
Thanks for the clarification, I updated the RAM threshold again ;)

Reply:
Updated again ;)